Sources/Tools/plShaderAssembler/main.cpp

```diff
@@ -117,14 +117,36 @@ void ICreateHeader(const ST::string& varName, const plFileName& fileName, FILE*
     ST::printf(fp, "\t0x{x},", *codes++);
     ST::printf(fp, "\t0x{x},", *codes++);
     ST::printf(fp, "\t0x{x}\n", *codes++);
-    fputs("\t};", fp);
-    fputs("\n\n", fp);
+    ST::printf(fp, "\t];");
+    ST::printf(fp, "\n\n");
 
     ST::printf(fp, "static const plShaderDecl {}Decl(\"{}\", {}, {}byteLen, {}Codes);\n\n",
                varName, fileName, varName, varName, varName);
     ST::printf(fp, "static const plShaderRegister {}Register(&{}Decl);\n\n", varName, varName);
 }
 
+static bool AttemptToOpenOutFile(plFileName outFile, hsUNIXStream& outFp)
+{
+	if (!outFp.Open(outFile, "w")) {
+		fputs("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n", stderr);
+		ST::printf(stderr, "Error opening file {} for output\n", outFile);
+		fputs("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n", stderr);
+		return true;
+	}
+	return false;
+}
+
+static bool AttemptToOpenInputFile(plFileName inFile, hsUNIXStream& inFp)
```

Comment:
```suggestion
static bool AttemptToOpenInputFile(const plFileName& inFile, hsUNIXStream& inFp)
```

No need to copy the file name. For bonus modernization, instead of returning a success value and taking a result reference, you could return an `std::optional<hsUNIXStream>`.